File: frontend/src/app/shared/dto/activeSubstationInputFlexibilityServiceSchedule-dto.ts
export class ActiveSubstationInputFlexibilityServiceScheduleDTO{
  //FIX ME
    public  inertiaEmulationPercentage:string = '';
    public  inertiaEmulation:any[] = [];
    public  frequencyRegulationPercentage:string= '';
    public  frequencyRegulation:any[] = [];
    public  modalFrequenciesPercentage:string= '';
    public  modalFrequencies:any[] = [];
    public  voltageRegulationPercentage:string= '';
    public  voltageRegulation:any[] = [];
    public  resourceVariabilityPercentage:string= '';
    public  resourceVariability:any[] = [];
    public  powerShiftingPercentage:string= ''; 
    public  powerShifting:any[] = [];
    public  rampRatePercentage:string= '';
    public  rampRate:any[] = [];
    public  powerCurtailmentPercentage:string= '';
    public  powerCurtailment:any[] = [];
    public  powerBalancePercentage:string= '';
    public  powerBalance:any[] = [];
    public  congestionReliefPercentage:string= '';
    public  congestionRelief:any[] = [];
    public  powerBalanceReserve: boolean
}

File: backend/src/main/java/crm/cloudApp/backend/dto/data/modules/active_substation/input/ActiveSubstationInputFlexibilityServiceScheduleDTO.java
package crm.cloudApp.backend.dto.data.modules.active_substation.input;

import crm.cloudApp.backend.dto.common.BaseDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.hibernate.annotations.CacheConcurrencyStrategy;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import java.util.List;

@Data
@DynamicUpdate
@DynamicInsert
@NoArgsConstructor
@AllArgsConstructor
@org.hibernate.annotations.Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
@Accessors(chain = true)
@Builder
public class ActiveSubstationInputFlexibilityServiceScheduleDTO extends BaseDTO {

    private List<Integer> inertiaEmulation;
    private List<Integer> frequencyRegulation;
    private List<Integer> modalFrequencies;
    private List<Integer> voltageRegulation;
    private List<Integer> resourceVariability;
    private List<Integer> powerShifting;
    private List<Integer> rampRate;
    private List<Integer> powerCurtailment;
    private List<Integer> powerBalance;
    private List<Integer> congestionRelief;
    private List<Integer> powerBalanceReserveUp;
    private List<Integer> powerBalanceReserveDown;


}
